zerver/views/presence.py

```diff
@@ -41,8 +41,8 @@ def get_presence_backend(request, user_profile, email):
     # For initial version, we just include the status and timestamp keys
     result = dict(presence=presence_dict[target.email])
     for val in result['presence'].values():
-        del val['client']
-        del val['pushable']
+        val.pop('client', None)
+        val.pop('pushable', None)
     return json_success(result)
```

Comment:
So the other thing that we need to change, probably just for this endpoint, and this should probably be a second commit, is that regardless of the listed idle status, if a user's timestamp is more than 5 minutes old, we should report that user as "offline", not "idle" or "online", and the timestamp then is when you were last online.

Basically, "idle" is meant to be "you have a connected client, but aren't using it", while "online" is "you have a connected client, and are using it".

Does that make sense?